chore: add changeset guidelines (#2361)
toptal/picasso: docs/contribution/changeset-guidelines.md

# Changesets guidelines

## Version bump
`patch` - a bug fix, a slight change in the existing functionality or the change, which doesn’t affect Picasso users, but good to know it exists.

> `minor` - a new prop, new functionality for the component. Picasso users will be benefited if know this change has been made.

Reviewer: Okay :) Here is how it is confusing to me. 
* `for the component` - If not component? What if functionality is in hook?
* `Picasso users will be benefited if know this change has been made.` technically using passive voice is bad in docs. Now, the meaning I can't get. What benefit? What if change exists and I don't know? I will not benefit?

TBH to me it was super clear to know that minor is `New functionality that is not breaking change`. With :point_up: I am confused about benefits, `if` conditions and `know` vs `not know`.

> `minor` - if you’ve added a new prop, new functionality for the component, which didn’t exist before. Picasso users will be benefited if would know that this change has been made.

Reviewer: what if we remove most of the words? Will it still contain same info? :) 
```suggestion
`minor` - prop or functionality added. Users will benefited of knowing about the change.
```
Not sure what is the value of this phrase `Users will benefited of knowing about the change.` if we do release it is a benefit anyway, why saying about it? :man_shrugging:

Author: @OleksandrNechai did some changes, please take a look 🙇

`major` - if you’ve changed existing prop values, removed the prop, or changed the purpose of the prop. Also, if you’ve affected the horizontal layout of the component (which can mean a broken layout of the page for users of this component), If you removed a component. Picasso users must know about this change because they need to do some actions to upgrade their existing Picasso version to the current one.

## Summary of the change (changelog message)

Please use this format:

```
---
'@toptal/[package]': [version bump]
---

> ---

Author: this is a problematic part. we need to have this additional `---` to display properly changelog instead of
<img width="390" alt="Screenshot 2021-12-30 at 17 31 48" src="https://user-images.githubusercontent.com/2836281/147765608-f5b27510-cc35-4e14-966f-4fb20f1f6056.png">

# Changesets guidelines

## Version bump
`patch`

- Bug fix
- Change in the existing functionality
- Change, which does not affect Picasso users

`minor`

- New property
- New property value
- New functionality
- New component

`major`

- Deleting a component
- Deleting a component property
- Changing values for existing properties
- Changing purpose of the property

> - Changing horizontal layout of a component (which can mean page layout brake where the component is used)

Reviewer: ```suggestion
- Changing horizontal layout of a component (which can mean page layout break where the component is used)
```
